libbpfgo.go

```diff
@@ -2089,3 +2089,11 @@ func BPFProgramTypeIsSupported(progType BPFProgType) (bool, error) {
 	}
 	return cSupported == 1, nil
 }
+
+func NumPossibleCPUs() (int, error) {
+	numCPUs, _ := C.libbpf_num_possible_cpus()
+	if numCPUs < 0 {
+		return 0, fmt.Errorf("failed to retrieve the number of CPUs")
```

Comment:
Sorry for the late review.

Shouldn't this error have errno information? For example:

https://github.com/aquasecurity/libbpfgo/blob/7c15abd453664293a200801392bfcc87162fe5b4/libbpfgo.go#L928-L930C3

Reply:
No worries at all.

Reply:
> I don't think we need to check for errno here, the only error path from the function is [here](https://github.com/libbpf/libbpf/blob/f7eb43b90f4c8882edf6354f8585094f8f3aade0/src/libbpf.c#LL12631C3-L12631C9), but let me know if you think that's not correct

We standardized returns taking into account possible changes in libbpf. Perhaps currently libbpf does not have a clear set of errno in this function, but it may be different in the future. And above all, as cgo provides us errno in C calls is always good to make use of it.